dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/search/TestSolrClientFactory.java

```diff
@@ -40,22 +42,28 @@ public SolrClient produceSolrClient() throws IOException {
     @Specializes
     @RorSolrClient
     public SolrClient produceRorSolrClient() {
-        String urlString = "http://localhost:" + resolveSolrPort() + "/solr/rorSuggestions";
+        String urlString = resolveSolrURL() + "/solr/rorSuggestions";
         LOGGER.fine("Creating test SolrClient at url: " + urlString);
 
         return new HttpSolrClient.Builder(urlString).build();
     }
-    
+
+
     public void disposeSolrClient(@Disposes SolrClient solrClient) throws IOException {
         solrClient.close();
     }
 
     // -------------------- PRIVATE --------------------
 
-    private static String resolveSolrPort() {
-        String port = System.getProperty("test.solr.port");
-        if (port == null) {
-            return DEFAULT_SOLR_TEST_PORT;
+    private static String resolveSolrURL() {
+        return "http://" + getPropertyOrDefault("test.solr.host", DEFAULT_SOLR_TEST_HOST) + ":" +
```

Comment:
How often are properties going to be changed during app runtime? Does this need to be done in method? If not, it would be nicer to set a constant in constructor once and then just access it.

Reply:
No, they won't change during tests. Moved URL initializations to the constructor.